phase_teach.rb

```diff
@@ -0,0 +1,37 @@
+module PhaseTeach
+
+  def set_phase(num)
+    @phase = num
+    "Cool, I've always wanted to teach phase #{num}!"
+  end
+
+  def teach_stuff
+    response = ""
+    if self.class == Teacher
+      response += "Listen, class, this is how everything works, fo SHO! "
+      response += "*drops flat-out insane knowledge bomb* "
+      response += "... You're welcome. *saunters away*"
+      response
+    elsif self.class == ApprenticeTeacher
+      response += "Listen, class, this is how everything works. "
+      response += "*drops crazy knowledge bomb* "
+      response += "... You're welcome."
+      response
+    end
+  end
+
+  def set_performance_rating(rating)
+    response = ""
+    if rating > 80 && self.class == ApprenticeTeacher
+      response = "Yay, I'm a great employee!"
+      receive_raise(@target_raise)
+    elsif rating > 90 && self.class == Teacher
```

Comment:
Any time you find yourself checking `self.class` manually against every class that might include you, that's a red flag.  

If a new file is created and needs to include your module, you have to modify the module to accommodate it!  Ideally your module should be plug-and-play, no modification needed.  So make a constant in each class and refer to the constant here.  Then if you add a new class, it's easy -- you just define the constant for that class, include the module, and you're done.